Polish block UI, inspector, add contributors, fix quote styles

jasmussen opened the pull request:
This is sort of a kitchen sink PR to fix a lot of lingering little issues.

# Normalization

The biggest issues fixed here are the normalization of block paddings and margins between the title, the empty placeholder, and the below the placeholder elements. These now share basically the same styles, which means they behave identically at responsive breakpoints (fixing issues), but the normalization also makes the code easier to maintain. 

Steps to test: View the editor at multiple responsive breakpoints (resize the window), and verify that

- new post writing prompt has the same width, hover and non hover, as the title and inserter
- insert a text block, then verify the inserter and title has the same width and sizes

# Other

Mover and cog/trash elements now have the same dimensions, and fit nicely within the smallest block (a single empty paragraph). This means the mover up/down arrows are spaced a little farther apart, but it gives some nice symmetry. Verify it looks good ;)

Quote styles were fixed, there was a CSS bleed regression from a recent tinymce update. To test, verify lineheights and margins for anything that uses the Editable component didn't regress from this fix. 

The inspector was polished a bit to remove double borders, tweak the close button padding, and add a white background when no block is selected.

The contributors document was updated with recent contributors. I know it's a bummer that this isn't automated, but it means we can give credit to contributors who test, mock up and other things.

jasmussen committed:
Update contributors list.

This will be an ongoing task. Anyone should feel free to add.

jasmussen committed:
Rewrite the responsive code of the title field.

This makes it the same as that of the block code, which fixes issues, and makes it more maintainable.

jasmussen committed:
Normalize and simplify the writing prompt responsive code.

This makes it the same as the block code, making it easier to maintain and fixing issues.

jasmussen committed:
Fix regression with quote styles.

jasmussen committed:
Polish the inspector tabbar.

Some pixels here, a few there. A background for when no block is selected. All good.

jasmussen committed:
Normalize paddings and margins in mover and settings.

This makes the mover (left) and cog/trash button (right) have the same margins and paddings, simplifying the appearance a bit.

jasmussen committed:
Normalize the completely empty writing prompt.

Like with the other prompts and titles, this rewrites the responsive code so it is easier to maintain (same as elsewhere), and fixes issues.

mtias commented:
On full-width blocks there's a slight overlap with the border:

![image](https://user-images.githubusercontent.com/548849/28669556-00c332a6-72d5-11e7-8284-fdecb563c47f.png)

mtias reviewed (approved):
Outside of that tiny detail, looks good to me.

jasmussen committed:
Fix positioning of cog and trash on full-wide images.

jasmussen committed:
Surface mover in fullwide also.

Because why not.

jasmussen commented:
Addressed the feedback, and also surfaced the mover in fullwide. Because why not. 

![screen shot 2017-07-27 at 14 22 08](https://user-images.githubusercontent.com/1204802/28670019-2303efde-72d7-11e7-9d24-4bc97fccf3a2.png)

Merging when the checks pass.

jasmussen commented:
Rebased. Will wait again.

mtias commented:
> Addressed the feedback, and also surfaced the mover in fullwide. Because why not.

Yes, good call.

mtias commented:
On some breakpoints the arrows overlap:

![image](https://user-images.githubusercontent.com/548849/28670458-0c703870-72d9-11e7-83fe-4a5ab2db2a89.png)

Also it removes the mobile toolbar for wide blocks :)

![image](https://user-images.githubusercontent.com/548849/28670485-2576c258-72d9-11e7-8b5b-85ba65e32788.png)

jasmussen commented:
Ack, need to revisit that. Okay.

jasmussen committed:
Address feedback.

Fix issue with movers overlapping, and with mobile toolbar being broken on wide.

jasmussen committed:
Polish the "continue writing" prompt.

jasmussen committed:
Move the z-index for movers lower.

I couldn't think of any side-effects to this.

jasmussen committed:
Fix issues with gallery block hoz scrollbar.

Only showed when settings was closed.

mtias commented:
Looks good to me now.

jasmussen commented:
Thanks. Merging. Travis needs work anyway.

jasmussen commented:
Okay maybe I should have waited for the tests to finish, my apologies. I was under the impression travis was totally broken, when in fact it's probably only partially broken.
